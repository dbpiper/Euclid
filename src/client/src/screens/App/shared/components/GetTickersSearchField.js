import React from 'react';
import _ from 'lodash';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import PropTypes from 'prop-types';
import ErrorBoundary from 'react-error-boundary';
import SearchField from './SearchField';

const categories = [
  { value: 'stocks', label: 'Stocks' },
];

export const GET_TICKERS_QUERY = gql`
  query {
    tickers
  }
`;

class GetTickersSearchField extends React.Component {
  constructor() {
    super();
    this.state = {
      selectedSearchItem: null,
      selectedCategory: null,
    };
  }

  handleChangeCategory = (selectedCategory) => {
    this.setState({ selectedCategory });
  };

  handleChangeSearchItem = (selectedSearchItem) => {
    const { onTickerSelect } = this.props;
    this.setState({ selectedSearchItem });
    onTickerSelect(selectedSearchItem.value);
  };

  render() {
    const { selectedSearchItem, selectedCategory } = this.state;
    return (
      <ErrorBoundary>
        <Query
          query={GET_TICKERS_QUERY}
        >
          {({
            loading,
            error,
            data,
          }) => {
            if (loading && error === false) return <p>Loading...</p>;
            if (error) return <p>Error :(</p>;
            const tickerOptions = _.map(data.tickers, ticker => (
              { value: ticker, label: ticker }
            ));
            return (
              <ErrorBoundary>
                <SearchField
                  options={tickerOptions}
                  categories={categories}
                  handleChangeCategory={this.handleChangeCategory}
                  handleChangeSearchItem={this.handleChangeSearchItem}
                  selectedCategory={selectedCategory}
                  selectedSearchItem={selectedSearchItem}
                />
              </ErrorBoundary>
            );
          }}
        </Query>
      </ErrorBoundary>
    );
  }
}

GetTickersSearchField.propTypes = {
  onTickerSelect: PropTypes.func.isRequired,
};

export default GetTickersSearchField;
