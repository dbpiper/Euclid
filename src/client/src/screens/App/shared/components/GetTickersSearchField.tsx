import gql from 'graphql-tag';
import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import { Query } from 'react-apollo';
import ErrorBoundary from 'react-error-boundary';
import SearchField from './SearchField';

const categories = [
  { value: 'stocks', label: 'Stocks' },
];

export const GET_TICKERS_QUERY = gql`
  query {
    tickers
  }
`;

class GetTickersSearchField extends React.Component<any, any> {
  public static propTypes: any;

  constructor(props: any) {
    super(props);
    this.state = {
      selectedSearchItem: null,
      selectedCategory: null,
    };
  }

  public handleChangeCategory = (selectedCategory: any) => {
    this.setState({ selectedCategory });
  }

  public handleChangeSearchItem = (selectedSearchItem: any) => {
    const { onTickerSelect } = this.props;
    this.setState({ selectedSearchItem });
    onTickerSelect(selectedSearchItem.value);
  }

  public render() {
    const { selectedSearchItem, selectedCategory } = this.state;
    return (
      <ErrorBoundary>
        <Query
          query={GET_TICKERS_QUERY}
        >
          {({
            loading,
            error,
            data,
          }) => {
            if (loading && typeof error === 'undefined') return <p>Loading...</p>;
            if (error) return <p>Error :(</p>;
            const tickerOptions = _.map(data.tickers, ticker => (
              { value: ticker, label: ticker }
            ));
            return (
              <ErrorBoundary>
                <SearchField
                  options={tickerOptions}
                  categories={categories}
                  handleChangeCategory={this.handleChangeCategory}
                  handleChangeSearchItem={this.handleChangeSearchItem}
                  selectedCategory={selectedCategory}
                  selectedSearchItem={selectedSearchItem}
                />
              </ErrorBoundary>
            );
          }}
        </Query>
      </ErrorBoundary>
    );
  }
}

GetTickersSearchField.propTypes = {
  onTickerSelect: PropTypes.func.isRequired,
};

export default GetTickersSearchField;
