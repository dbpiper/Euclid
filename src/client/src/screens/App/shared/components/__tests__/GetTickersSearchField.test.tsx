import { shallow } from 'enzyme';
import React from 'react';
import GetTickersSearchField from '../GetTickersSearchField';

let component: any;

// tslint:disable-next-line no-empty
const nothing = () => {};

beforeEach(() => {
  component = shallow(
    <GetTickersSearchField
      onTickerSelect={nothing}
    />,
  );
});

test('GetTickersSearchField basic', () => {
  expect(component).toMatchSnapshot();
});
