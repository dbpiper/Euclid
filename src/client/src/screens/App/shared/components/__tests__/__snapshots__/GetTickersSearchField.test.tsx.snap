// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetTickersSearchField basic 1`] = `
<ErrorBoundary
  FallbackComponent={[Function]}
>
  <TickersQuery
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "Tickers",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "tickers",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 35,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </TickersQuery>
</ErrorBoundary>
`;
